src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -231,7 +231,7 @@ sealed class HashMap[A, +B] extends AbstractMap[A, B]
               result = result.updated0(key, computeHash(key), 0, kv._2, kv, merger)
             }
           }
-          that.asInstanceOf[scala.Traversable[(A,B)]] foreach adder
+          that.asInstanceOf[GenTraversableOnce[(A,B)]] foreach adder
```

Comment:
not sure here, but it seemed a bit strange for this code to be different than `++` above